app/search.tsx

```diff
@@ -0,0 +1,433 @@
+          </View>
+        )}
+
+        {showSearchResults && (
+          <FlatList
+            data={searchData?.movies || []}
+            renderItem={renderSearchResultItem}
+            keyExtractor={item => item.id.toString()}
+            style={styles.searchResultsList}
+            contentContainerStyle={styles.searchResultsContent}
+          />
+        )}
+
+        {showEmptyState && (
+          <View style={styles.emptyContainer}>
+            <ThemedText style={styles.emptyText}>
+              No movies found for "{debouncedQuery}"
+            </ThemedText>
+          </View>
+        )}
+      </View>
+    </ThemedSafeAreaView>
+  )
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  header: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    paddingHorizontal: 20,
+    paddingVertical: 12,
+    gap: 12,
+  },
+  backButton: {
+    padding: 4,
+  },
+  searchInputContainer: {
+    flex: 1,
+    flexDirection: 'row',
+    alignItems: 'center',
+    borderRadius: 12,
+    borderWidth: 1,
+    paddingHorizontal: 12,
+    height: 44,
+  },
+  searchIcon: {
+    marginRight: 8,
+  },
+  searchInput: {
+    flex: 1,
+    fontSize: 16,
+    padding: 0,
+  },
+  filterButtonInside: {
+    padding: 4,
+    marginLeft: 8,
+  },
+  content: {
+    flex: 1,
+    paddingHorizontal: 20,
+  },
+  recentSearchesSection: {
+    flex: 1,
+  },
+  sectionHeader: {
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    marginBottom: 16,
+    marginTop: 8,
+  },
+  sectionTitle: {
+    fontSize: 18,
+    fontWeight: '600',
+  },
+  clearAllText: {
+    fontSize: 14,
+    color: '#FF3B30',
+    fontWeight: '500',
+  },
+  recentSearchesList: {
+    flex: 1,
+  },
+  recentSearchesListContent: {
+    paddingRight: 20,
+  },
+  recentSearchCard: {
+    marginRight: 16,
+    width: 200,
+  },
+  recentSearchPosterContainer: {
+    position: 'relative',
+    width: 200,
+    height: 250,
+    borderRadius: 32,
+    overflow: 'hidden',
+    marginBottom: 8,
+  },
+  recentSearchPoster: {
+    width: '100%',
+    height: '100%',
+    borderRadius: 32,
+  },
+  placeholderImage: {
+    backgroundColor: '#2C2C2E',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  recentSearchPlayOverlay: {
+    position: 'absolute',
+    bottom: 12,
+    right: 12,
+    width: 40,
+    height: 40,
+    backgroundColor: '#1E1E1E',
+    borderRadius: 20,
+    justifyContent: 'center',
+    alignItems: 'center',
+    opacity: 0.9,
+  },
+  recentSearchCardTitle: {
+    fontSize: 16,
+    color: '#FFFFFF',
+    textAlign: 'center',
+    fontWeight: '500',
+    paddingHorizontal: 4,
+  },
+  loadingContainer: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingTop: 40,
+  },
+  errorContainer: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    paddingTop: 40,
+  },
+  errorText: {
+    color: '#FF3B30',
```

Comment:
Check if we already have a color for the errors. If not Add the colors to the colors constants.